### ISSUE TITLE
typing.overload on `__call__`  makes class not callable

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1392]
I made a callable class `Foo`, but pyright failed to infer it is callable:

```python
from typing import Callable, overload

class Foo:
    @overload
    def __call__(self, x: int) -> int:
        ...

    @overload
    def __call__(self, x: str) -> str:
        ...

    def __call__(self, x):
        return x * 2


foo: Callable = Foo() # ERROR
```

Error:

```
  17:17 - error: Expression of type "Foo" cannot be assigned to declared type "(*args: Unknown, **kwargs: Unknown) -> Unknown"
    Type "Foo" cannot be assigned to type "(*args: Unknown, **kwargs: Unknown) -> Unknown" (reportGeneralTypeIssues)
```

When I removed overloaded methods, it works:

```python
class Foo:
    def __call__(self, x):
        return x * 2


foo: Callable = Foo() # No ERROR
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.104, which I just published. It will also be included in the next release of pylance.
